crates/holochain/src/core/workflow/app_validation_workflow.rs

```diff
@@ -303,6 +312,27 @@ async fn app_validation_workflow_inner(
                 let awaiting_ops = awaiting_ops.clone();
                 let rejected_ops = rejected_ops.clone();
 
+                if let Outcome::Rejected(_) = &outcome {
+                    let warrant_op =
+                        crate::core::workflow::sys_validation_workflow::make_warrant_op(
+                            &conductor,
+                            &dna_hash,
+                            &chain_op,
+                            ValidationType::App,
+                        )
+                        .await?;
+
+                    workspace
+                        .authored_db
+                        .write_async(move |txn| {
+                            warn!("Inserting warrant op");
```

Comment:
I suggest to add more info to this warning. I'd actually merge it with the one below and output the op that was rejected and the warrant op.